### PR TITLE
Anmacdonald/authenticationview

### PR DESCRIPTION
**Pull request by anmacdonald**
AuthenticationView, C++/QML in new Toolkit style.

### CODE REVIEW

File: import/OAuth2View.qml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        url: controller.currentChallengeUrl
+
+        onLoadingChanged: {
+            if (loadRequest.status === WebView.LoadSucceededStatus) {
+                forceActiveFocus();
+            }
+        }
+
+        onTitleChanged: {
+            if (title.indexOf("SUCCESS code=") > -1) {
```
Comment (lsmallwood) [inline import/OAuth2View.qml:60]: this feels like business logic in the view

File: import/AuthenticationView.qml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    \li HTTP Basic (UserCredentialsView)
+    \li HTTP Digest (UserCredentialsView)
+    \li IWA (UserCredentialsView)
+    \li OAuth 2.0 (OAuth2View)
+    \li SAML (OAuth2View)
+    \li PKI (ClientCertificateView)
+    \li SSL Handshake Warnings (SslHandshakeView)
+  \endlist
+  
+  \note OAuth 2.0 uses a WebView. To use an OAuthView you must call
```
[the remark this answers is not in the document]
Comment (anmacdonald) [reply import/AuthenticationView.qml:44]: QML as well unforunately! I have tried.

File: import/OAuth2View.qml (hunk 1 above)
Comment (anmacdonald) [reply import/OAuth2View.qml:60]: In an ideal world we wouldn't be reading from the title field at all. It's an unfortunate holdover from the old toolkit OAuth. I'm not sure how to make it more robust.

**Comment by anmacdonald** [comment]
@ldanzinger Ready for docs review. I've fixed some doc error outside of AuthenticationView.

File: cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+  if (m_deleteChallengeOnProcessed) 
+    m_currentChallenge->deleteLater();
+
+  m_currentChallenge = nullptr;
+  emit challengeChanged();
+}
+
+/*!
+  \fn void Esri::ArcGISRuntime::Toolkit::AuthenticationController::challengeChanged()
+  \brief Emitted when the challenge changes.
```
Comment (ldanzinger) [inline cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp:303]: what does this mean? The challenge has been mutated in some way? Or that the current challenge of the authentication controller has changed to a different challenge? I wonder if a different signal name like currentChallengeChanged would be better?

File: cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+ */
+
+/*!
+  \fn void Esri::ArcGISRuntime::Toolkit::AuthenticationController::clientCertificateInfosChanged()
+  \brief Emitted when the clientCertificateInfos updates.
+  \sa clientCertificateInfosChanged
```
Comment (ldanzinger) [inline cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp:309]:
```suggestion
  \sa clientCertificateInfos
```

File: cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+/*!
+  \fn void Esri::ArcGISRuntime::Toolkit::AuthenticationController::clientCertificatePasswordRequired(QUrl certificate)
+  \brief Emitted when a \a certificate that was added to the AuthenticationManager requires a password.
+  \sa addClientCertificate
+ */
+
+/*!
+  \property Esri::ArcGISRuntime::Toolkit::AuthenticationController::currentChallengeUrl
+  \brief Url of current challenge.
```
Comment (ldanzinger) [inline cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp:320]: I'd recommend using some articles to make more complete sentences - 

```suggestion
  \brief The Url of the current challenge.
```

The next couple of descriptions have this styling as well

File: cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+  \brief Type of current challenge.
+ */
+
+/*!
+  \property Esri::ArcGISRuntime::Toolkit::AuthenticationController::currentChallengeFailureCount
+  \brief Failure count of current challenge.
+ */
+
+/*!
+  \property Esri::ArcGISRuntime::Toolkit::AuthenticationController::clientCertificateInfos
```
Comment (ldanzinger) [inline cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp:339]: does property not require you to specify type? it just figures out from the associated getter/setter?

File: import/AuthenticationController.qml [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+import Esri.ArcGISRuntime 100.8
+
+/*!
+  \qmltype AuthenticationController
+  \inqmlmodule Esri.ArcGISRuntime.Toolkit
+  \since Esri.ArcGISRutime 100.9
+  \brief In MVC architecture, this is the controller for the corresponding
+  AuthenticationView.  
+  
+  This controller is a thin wrapper around a AuthenticationManager. As
```
Comment (ldanzinger) [inline import/AuthenticationController.qml:27]:
```suggestion
  This controller is a thin wrapper around the AuthenticationManager. As
```

File: import/AuthenticationView.qml [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        instance is created. See \l{https://doc.qt.io/qt-5/qtwebview-index.html}
+        {Qt WebView}.
+ */
+
+Item {
+    id: authenticationView
+
+    /*!
+      \qmlproperty AuthenticationController controller.
+      \brief the Controller handles references to challenge semitted by the
```
Comment (ldanzinger) [inline import/AuthenticationView.qml:55]:
```suggestion
      \brief the Controller handles references to challenges emitted by the 
```

File: import/CoordinateConversion.qml [hunk 9]
```diff
@@ -20,17 +20,17 @@ import QtQuick.Controls 2.12
...
+ \qmltype CoordinateConversion
+ \inqmlmodule Esri.ArcGISRuntime.Toolkit
+ \since Esri.ArcGISRutime 100.8
+ \brief The user interface for the coordinate conversion tool.
+
+ This tool allows a user to select a point on the map or to enter a point by
+ text entry.
+ The point may be highlighted by zooming into that point, or by highlighting
+ the point on the GeoView.
+ A list of different coordinate formats represnenting the same point are
```
Comment (ldanzinger) [inline import/CoordinateConversion.qml:32]:
```suggestion
 A list of different coordinate formats representing the same point are
```

File: cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp (hunk 6 above)
Comment (anmacdonald) [reply cpp/Esri/ArcGISRuntime/Toolkit/AuthenticationController.cpp:339]: Not in the C++ case https://doc.qt.io/qt-5/qtwritingstyle-cpp.html#properties

**Pull request merged**